includes/vendor/mautic.php

```diff
@@ -0,0 +1,607 @@
+		'redirect_uri' => $credentials['callback'],
+		'sslverify' => false,
+		);
+		if ( 'authorization_code' == $grant_type ) {
+			$body['code'] = $credentials['access_code'];
+		} else {
+			$body['refresh_token'] = $credentials['refresh_token'];
+		}
+			// Request to get access token.
+			$response = wp_remote_post( $url, array(
+				'method' => 'POST',
+				'timeout' => 45,
+				'redirection' => 5,
+				'httpversion' => '1.0',
+				'blocking' => true,
+				'headers' => array(),
+				'body' => $body,
+				'cookies' => array(),
+				)
+			);
+		return $response;
+	}
+
+	/**
+	 * Add contacts to Mautic, Add to segments, return GET request data
+	 *
+	 * @since 1.0.0
+	 * @param string $url api endpoint.
+	 * @param string $method API menthod.
+	 * @param array  $param parameters.
+	 * @param array  $segments mautic segments ID.
+	 * @return void
+	 */
+	public static function ampw_mautic_api_call( $url, $method, $param = array(), $segments = array() ) {
+
+		$status = 'success';
+
+		self::generate_access_token();
+		// add contacts.
+		$credentials = APMautic_helper::get_mautic_credentials();
+
+		if ( ! isset( $credentials['access_token'] ) ) {
+			return;
+		}
+		$access_token = $credentials['access_token'];
+		$param['access_token'] = $access_token;
+
+		$url = $credentials['baseUrl'] . $url;
+
+		$param['ipAddress'] = $_SERVER['REMOTE_ADDR'];
+		$response = wp_remote_post( $url, array(
+			'method' => $method,
+			'timeout' => 45,
+			'redirection' => 5,
+			'httpversion' => '1.0',
+			'blocking' => true,
+			'headers' => array(),
+			'body' => $param,
+			'cookies' => array(),
+		));
+		if ( is_wp_error( $response ) ) {
+			$error_msg = $response->get_error_message();
+			$status = 'error';
+			echo __( 'There appears to be an error with the configuration.', 'automateplug-mautic-wp' );
+
+		} else {
+
+			if ( is_array( $response ) ) {
+
+				$response_code = wp_remote_retrieve_response_code( $response );
+
+				if ( 200 === $response_code || 201 === $response_code ) {
+
+					$response_body = wp_remote_retrieve_body( $response );
+					$contact_created = json_decode( $response_body );
+					$contact = $contact_created->contact;
+
+					if ( isset( $contact->id ) ) {
+
+						$contact_id = (int) $contact->id;
+						$res = self::contact_segment_subscribe( $contact_id, $credentials, $segments );
+						$status = $res['status'];
+					}
+				} else {
+					$ret = false;
+					$status = 'error';
+					$error_msg = isset( $response['response']['message'] ) ? $response['response']['message'] : '';
+				}
+			}
+		}
+	}
+
+	/**
+	 * Get Method and URL according to user email
+	 *
+	 * @since 1.0.0
+	 * @param string $email user email.
+	 * @return array $data
+	 */
+	public static function get_api_method_url( $email ) {
+
+		$credentials = APMautic_helper::get_mautic_credentials();
+		$data = array();
+		$contact_id = $email_cid = '';
+
+		if ( isset( $_COOKIE['mtc_id'] ) ) {
+
+			// for anonymous contacts.
+			$contact_id = $_COOKIE['mtc_id'];
```

Comment:
All the instances in the code where $_Cookie is accessed, it should be correctly escaped.